### PR TITLE
feat: keep sender in tree task

### LINKED ISSUE #9753: Support sending messages to EngineApiImpl from itself
currently the engineapi impl has an incoming channel on which it listens

https://github.com/paradigmxyz/reth/blob/70a0cea941c7049c488cd4cf5191c456f4b9f390/crates/engine/tree/src/tree/mod.rs#L381-L382

there are situations where it would be beneficial if the engine could send itself messages, to spread the workload across multiple actions, for example, when we need to close a gap by re-executing multiple buffered blocks, which could take a bit.

we can split the workload and prevent timeouts if these are split across multiple messages, so we can also abort mid request handling and continue doing that by sending a new message with the remaining blocks for example

# TODO
* wrap the incoming channel in a custom channel wrapper that is a superset of `FromEngine<BeaconEngineMessage<T>>` which is then returned to interact with this service but internally allows receiving additional events

### PR DESCRIPTION
**Pull request by mattsse**
closes https://github.com/paradigmxyz/reth/issues/9753

see docs why we need this.

TLDR, this makes it possible to split a larger range of blocks into multiple smaller packets, this is easier than limiting the range of blocks we download because we download in reverse.
and could have more use cases where we need to do async work.

splitting block ranges will happen in a followup, I think we can introduce a setting for how many blocks to execute sequentially

### CODE REVIEW

**Review by Rjected — APPROVED** [review]
makes sense to send remaining blocks over this channel
